public/app/features/alerting/unified/mocks/server/events.ts

```diff
@@ -21,3 +22,46 @@ export function waitForServerRequest(handler: HttpHandler) {
     });
   });
 }
+
+interface SerializedRequest {
+  method: string;
+  url: string;
+  body: string | JsonValue;
+  headers: string[][];
+}
+
+export async function captureRequests(): Promise<Request[]> {
```

Comment:
I'd be tempted to mark this as `@deprecated` with a note (like `waitForServerRequest`) just so we always consider whether we should be using it - even if it's never formally deprecated

Reply:
I think that's sensible, I can't think of any other keywords that would work here to discourage its usage :D